scalalib/src/TestModule.scala

```diff
@@ -0,0 +1,215 @@
+package mill.scalalib
+
+import mill.{Agg, BuildInfo, T}
+import mill.define.{Command, Task, TaskModule}
+import mill.eval.Result
+import mill.modules.Jvm
+
+trait TestModule extends JavaModule with TaskModule {
+  override def defaultCommandName() = "test"
+
+  /**
+   * What test frameworks to use.
+   */
+  @deprecated("Use testFramework instead.", "mill after 0.9.6")
+  def testFrameworks: T[Seq[String]] = T { Seq.empty[String] }
+
+  /**
+   * The test framework to use.
+   */
+  def testFramework: T[String] = T {
+    val frameworks = testFrameworks()
+    val msg =
+      "Target testFrameworks is deprecated. Please use target testFramework or use on of the " +
+        "predefined TestModules: TestNg, Junit, Scalatest, ..."
+    if (frameworks.size != 1) {
+      Result.Failure(
+        "Since mill after-0.9.6 only one test framework per TestModule is supported. " ++ msg)
```

Comment:
First time I see string concatenation with `++` :) Is it to not confuse it with the sum?

Reply:
I have no idea, haha. Will fix it...